File: python/src/Structural/Bridge/SuperSizeSoda.py

# two of two classes extending the Abstract

from Soda import Soda

class SuperSizeSoda(Soda):

	def __init__(self): 
		self.setSodaImp()

	def pourSoda(self) :
		sodaImp = self.getSodaImp()
		retval = []
		for i in range(5):
			retval.append('...glug...')
			retval.append(sodaImp.pourSodaImp())
		return retval

File: ruby/src/Structural/Bridge/SuperSizeSoda.rb

# - one of two classes extending the Abstract

require "Structural/Bridge/Soda"

class SuperSizeSoda < Soda

	def initialize() 
		setSodaImp()
	end

	def pourSoda()
 		sodaImp = getSodaImp()
		retval = []
		( 1..5 ).each do
			retval.push('...glug...')
			retval.push(sodaImp.pourSodaImp())
		end
		retval.join(' ')
	end

end

